src/internal/util/reportError.ts

```diff
@@ -0,0 +1,34 @@
+import { Subscriber } from '../Subscriber';
+import { rxSubscriber as rxSubscriberSymbol } from '../symbol/rxSubscriber';
+import { ErrorObserver } from '../types';
+
+/**
+ * Reports the error to the ErrorObserver unless the observer is closed or
+ * stopped - in which case, an alternative reporting mechanism is used.
+ * @param err the error to report
+ */
+export function reportError(err: any, observer: ErrorObserver<any>, report?: (err: any) => void): void {
+  if (canReportError(observer)) {
+     observer.error(err);
+  } else {
+    (report || consoleReportError)(err);
+  }
+}
+
+function canReportError(observer: ErrorObserver<any>): boolean {
+  const { closed, destination, isStopped } = observer as any;
+  if (closed || isStopped) {
+    return false;
+  } else if (destination instanceof Subscriber || destination[rxSubscriberSymbol]) {
+    return canReportError(destination);
+  }
+  return true;
+}
+
+function consoleReportError(err: any): void {
```

Comment:
`consoleWarn`, this is too confusing considering we have a `hostReportError` function in the library, which is different.